app/models/itt_provider_form.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+class IttProviderForm
+  include ActiveModel::Model
+
+  attr_accessor :trn_request, :itt_provider_enrolled, :itt_provider_name
+
+  validates :itt_provider_enrolled, inclusion: { in: %w[yes no] }
+  validates :itt_provider_name, presence: true, length: { maximum: 255 }, if: -> { itt_provider_enrolled == 'yes' }
+
+  def save
+    return false if invalid?
+
+    trn_request.update!(itt_provider_enrolled: itt_provider_enrolled, itt_provider_name: itt_provider_name)
```

Comment:
We should probably use `update` without the bang so as to keep with the expected behaviour of `save`.